File: Find-the-Index-of-the-First-Occurrence-in-a-String.java
class Solution 
{
    public int strStr(String h, String n) 
    {
        int hsize = h.length();
        int nsize = n.length();

        if (nsize == 0) 
        {
            return 0;
        }

        for (int i = 0; i <= hsize - nsize; i++) 
        {
            if (h.substring(i, i + nsize).equals(n)) 
            {
                return i;
            }
        }

        return -1;
    }
}

File: Find-the-Index-of-the-First-Occurrence-in-a-String.cpp
class Solution {
public:
    int strStr(string h, string n) {
        int hsize = h.size();
        int nsize = n.size();

        if (nsize == 0){
            return 0;
        } 

        for (int i = 0; i <= hsize - nsize; i++) 
        {
            if (h.substr(i, nsize) == n) 
            {
                return i;
            }
        }

        return -1;
    }
};
